glean-core/ios/Glean/Glean.swift

```diff
@@ -152,7 +152,10 @@ public class Glean {
     /// If the ping currently contains no content, it will not be assembled and
     /// queued for sending.
     func sendPingsByName(pingNames: [String]) {
-        _ = Dispatchers.shared.launch {
+        // This runs in the background to allow for processing of pings off of the
+        // main thread. Please note that the ping uploader will spawn other async
+        // operations if there are pings to upload.
+        Dispatchers.shared.launchAsync {
```

Comment:
nit: maybe this could be just `launch`?

Reply:
The bug was to move this off of the serial queue.  If we call this with `launch` and it takes some time to perform the network upload, then the serial execution queue is blocked.

Reply:
The `launchAsync` puts this on a different execution queue that is concurrently executed so as to not block the execution of other tasks.